crates/ironrdp-dvc-pipe-proxy/src/platform/windows/worker.rs

```diff
@@ -0,0 +1,173 @@
+    let WorkerCtx {
+        mut pipe,
+        to_pipe_rx,
+        to_pipe_semaphore,
+        abort_event,
+        dvc_write_callback,
+        pipe_name,
+        channel_name,
+        channel_id,
+    } = worker_ctx;
+
+    info!(%channel_name, %pipe_name, "Connecting DVC pipe proxy");
+
+    {
+        let mut connect_ctx = pipe.prepare_connect_overlapped()?;
+
+        if !connect_ctx.overlapped_connect()? {
+            const EVENT_ID_ABORT: usize = 0;
+            let events = &[abort_event.raw(), connect_ctx.event().raw()];
+
+            let wait_result = match wait_any_with_timeout(events, PIPE_CONNECT_TIMEOUT) {
+                Ok(idx) => idx,
+                Err(WindowsError::WaitForMultipleObjectsTimeout) => {
+                    warn!(%channel_name, %pipe_name, "DVC pipe proxy connection timed out");
+                    return Ok(());
+                }
+                Err(err) => {
+                    return Err(DvcPipeProxyError::Windows(err));
+                }
+            };
+
+            if wait_result == EVENT_ID_ABORT {
+                info!(%channel_name, %pipe_name, "DVC pipe proxy connection has been aborted");
+                return Ok(());
+            }
+
+            connect_ctx.get_result()?;
+        }
+    }
+
+    info!(%channel_name, %pipe_name, "DVC pipe proxy connected");
+
+    let mut read_ctx = pipe.prepare_read_overlapped(MESSAGE_BUFFER_SIZE)?;
+
+    const EVENT_ID_ABORT: usize = 0;
+    const EVENT_ID_READ: usize = 1;
+    const EVENT_ID_WRITE_MPSC: usize = 2;
+    let events = &[abort_event.raw(), read_ctx.event().raw(), to_pipe_semaphore.raw()];
+
+    read_ctx.overlapped_read()?;
+
+    info!(%channel_name, %pipe_name, "DVC pipe proxy IO loop started");
+
+    loop {
+        let wait_result = wait_any(events)?;
+
+        // abort event
+        if wait_result == EVENT_ID_ABORT {
+            info!(%channel_name, %pipe_name, "DVC pipe proxy connection has been aborted");
+            return Ok(());
+        }
+
+        // read from pipe
+        if wait_result == EVENT_ID_READ {
+            let read_result = read_ctx.get_result()?.to_vec();
+
+            trace!(%channel_name, %pipe_name, "DVC proxy read {} bytes from pipe", read_result.len());
+
+            if !read_result.is_empty() {
+                let messages = encode_dvc_messages(
+                    channel_id,
+                    vec![Box::new(RawDataDvcMessage(read_result))],
+                    ChannelFlags::empty(),
+                )
+                .map_err(|_| DvcPipeProxyError::EncodeDvcMessage)?;
+
+                if let Err(err) = dvc_write_callback(0, messages) {
+                    error!(%err, %channel_name, %pipe_name, "DVC pipe proxy write callback failed");
+                }
+            }
+
+            // Queue the read operation again
+            read_ctx.overlapped_read()?;
+            continue;
+        }
+
+        // read from mpsc and write to pipe
```

Comment:
style: https://github.com/Devolutions/IronRDP/blob/master/STYLE.md#inline-code-comments-are-proper-sentences

Try to keep this in mind:
- It’s easier to write useful comments
- It’s easier to not write unnecessary comments